src/TropicalGeometry/linear_space.jl

```diff
@@ -335,13 +349,72 @@ end
 #
 ###############################################################################
 
+# helper function to compute pluecker information of a tropical linear space if not available
+function ensure_pluecker_data!(TropL::TropicalLinearSpace)
+
+    if has_attribute(TropL,:tropical_pluecker_vector)
+        return
+    end
+
+    # check if tropical semiring map is available
+    if has_attribute(TropL,:tropical_semiring_map)
+        # if yes, we compute the tropical pluecker vector from the algebraic data
+        nu = tropical_semiring_map(TropL)
+
+        # check if algebraic pluecker vector is available
+        # if yes, compute the tropical pluecker vector
+        if has_attribute(TropL,:algebraic_pluecker_vector)
+            algebraicPlueckerVector = get_attribute(TropL,:algebraic_pluecker_vector)
+            tropicalPlueckerVector = nu.(algebraicPlueckerVector)
+            set_attribute!(TropL,:tropical_pluecker_vector,tropicalPlueckerVector)
+            return
+        end
+        # check if algebraic matrix is available
+        # if yes, compute the algebraic pluecker vector and rerun
+        if has_attribute(TropL,:algebraic_matrix)
+            A = get_attribute(TropL,:algebraic_matrix)
+            n = ncols(A)
+            k = rank(A)
+            plueckerIndices = AbstractAlgebra.combinations(1:n,k)
+            algebraicPlueckerVector = AbstractAlgebra.minors(A,k)
+            set_attribute!(TropL,:pluecker_indices,plueckerIndices)
+            set_attribute!(TropL,:algebraic_pluecker_vector,algebraicPlueckerVector)
+            return ensure_pluecker_data!(TropL)
+        end
+        # check if algebraic ideal is available
+        # if yes, compute the algebraic matrix and rerun
+        if has_attribute(TropL,:algebraic_ideal)
+            I = get_attribute(TropL,:algebraic_ideal)
+            A = matrix([[coeff(g,xi) for xi in gens(base_ring(I))] for g in gens(I)])
+            set_attribute!(TropL,:algebraic_matrix,A)
+            return ensure_pluecker_data!(TropL)
+        end
+    else
+        # if not, we have to compute the tropical pluecker vector from the tropical data
+        if has_attribute(TropL,:tropical_matrix)
```

Comment:
I find this function a bit difficult to untangle and I am not sure it is is always used right (see my other comments). In any case, isn't it true that if `tropical_semiring_map` is present, but none of `algebraic_pluecker_vector`, `algebraic_matrix`, `algebraic_ideal` are, then this method will fail -- even if `tropical_matrix` *is* available and thus would allow for the requested computation?

Reply:
Good catch, that is very true and definitely shouldn't be the case!

Reply:
Removing the `else` should do the trick, because if the data has been computed, we would have exited the function already.